unlock_indicator.c

```diff
@@ -678,6 +685,19 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
     cairo_surface_t *xcb_output = cairo_xcb_surface_create(conn, bg_pixmap, vistype, resolution[0], resolution[1]);
     cairo_t *xcb_ctx = cairo_create(xcb_output);
 
+    /*update image according to the slideshow_interval*/
+    if (slideshow_image_count > 0) {
+        unsigned long now = (unsigned long)time(NULL);
+        if (-1 == lastCheck || now - lastCheck >= slideshow_interval) {
+            img = img_slideshow[current_slideshow_index++];
```

Comment:
maybe allow for random indexing (i.e. use rand() to generate an index from [0, slideshow_image_count) )